client/my-sites/themes/theme-options.js

```diff
@@ -89,6 +93,16 @@ const tryandcustomize = {
 	hideForTheme: ( state, theme, siteId ) => isActive( state, theme.id, siteId )
 };
 
+const tryAndCustomizeOnJetpack = {
+	label: i18n.translate( 'Try & Customize' ),
+	header: i18n.translate( 'Try & Customize on:', {
+		comment: 'label in the dialog for opening the Customizer with the theme in preview'
+	} ),
+	action: installWpcomThemeAndTryAndCustomize,
+	hideForSite: ( state, siteId ) => ! canCurrentUser( state, siteId, 'edit_theme_options' ),
```

Comment:
I think we should also hide if `! isJetpackSite( state, siteId )`, just to be sure.